src/components/Common.jsx

```diff
@@ -79,7 +79,7 @@ const SystemInfoContextWrapper = ({ children, conf, osRelease }) => {
     );
 };
 
-const MaybeBackdrop = ({ children }) => {
+export const useMaybeBackdrop = () => {
     const [hasDialogOpen, setHasDialogOpen] = useState(false);
 
     useEffect(() => {
```

Comment:
I find this super weird. On the outside this is now a simple query function which returns a class name. But each time this gets called by the storage component (i.e. on each render), this will set up a new useEffects(). This doesn't feel healthy, and it feels like that effect should be pulled out of this function.

I already hate useEffects(), but this feels too inflationary.

Reply:
useMaybeBackdrop is a custom hook (https://react.dev/reference/rules/rules-of-hooks) with no dependencies, so it will run its useEffect only once, no matter how many re-renders the parent does.

I think actually this usage of hooks as I do it here, is the idea of custom hooks.

Btw I verified my thoughts by putting a console log inside the useMaybeBackdrop useEffect, and it indeed gets called only once.

Let's discuss this over chat if  you still disagree.

Reply:
In any case, because we dont need to wrap the whole app now with that class, I can extract the logic into the CockpitStorageIntergration component itself, and eliminate the hook if you prefer.

Reply:
@martinpitt btw this is the same exact concept as https://github.com/cockpit-project/cockpit/blob/main/pkg/lib/hooks.ts#L65 for example.

Reply:
OK! effects keep confusing me :cry: